ckanext/data_qld/reporting/logic/action/get.py

```diff
@@ -655,16 +655,16 @@ def overdue_datasets(context, data_dict):
     check_org_access(org_id, permission)
 
     try:
-        # next_update_due is stored as UTC without timezone as isoformat
-        today_utc = datetime.utcnow().date().isoformat()
+        # next_update_due is stored as display timezone without timezone as isoformat
+        today = datetime.now(h.get_display_timezone()).date().isoformat()
```

Comment:
Normally this is converted upon display. Why not do that here? Storing it as a specific timezone is likely to cause trouble if anyone from a different timezone views it. What was wrong with UTC?

Reply:
@ThrawnCA Some background information on this issue and field. 
The `next_update_due` field is a date picker on the dataset form where a user can select any date in the future and it is stored in iso format (yyyy-mm-dd) with no time or timezone information. Because there is no time or timezone information this field gets rendered as-is with no conversion.

This action is used for the admin report to retrieve datasets where their `next_update_due` is less than or equal to today's date which would make them `overdue`.
I will try to use an example to describe the issue found in UAT to help explain why it was updated this way.
A user creates a dataset called `Dataset A` and sets the `next_update_due` field to 2021-10-15 (tomorrow).
An admin user runs the Admin report tomorrow (2021-10-15) anytime between 00:00-09:59 am Brisbane time, the code will get the UTC date which is still 2021-10-14 because of the timezone difference (Brisbane is +10 UTC) hours and the dataset `Dataset A` will not be included in the report.
An admin user runs the Admin report tomorrow anytime (2021-10-15) after 10:00 am Brisbane time, the code will get the utc date which is now 2021-10-15  and `Dataset A` will be included in the report.
This makes it confusing to the user and is why it failed UAT because it should be included as `overdue` in `Australian/Brisbane` timezone.

Hopefully, that explains why we cannot use the UTC date for this query and it needs to be the `Australian/Brisbane` timezone.
Open to any suggestions to get around this issue but don't want to make this any more complicated than it needs to be.

cc. @duttonw

Reply:
@ThrawnCA this is not about saving to database, this is working which datasets are due based on the reporter who is generating it.

they did have utc time but that was not working before 10am brisbane. So using display time zone should ensure it is correct.

Reply:
Ah, if the generation here is for output rather than storage, that makes sense.